x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -98,81 +96,111 @@ public record UnresolvedPolicy(String name, Enrich.Mode mode) {
     /**
      * Resolves a set of enrich policies
      *
-     * @param targetClusters     the target clusters
+     * @param targetClusters     the target clusters; key: cluster alias; value: skip_unavailable setting (null for local cluster)
      * @param unresolvedPolicies the unresolved policies
      * @param listener           notified with the enrich resolution
      */
     public void resolvePolicies(
-        Collection<String> targetClusters,
+        Map<String, Boolean> targetClusters,
         Collection<UnresolvedPolicy> unresolvedPolicies,
         ActionListener<EnrichResolution> listener
     ) {
         if (unresolvedPolicies.isEmpty() || targetClusters.isEmpty()) {
             listener.onResponse(new EnrichResolution());
             return;
         }
-        final Set<String> remoteClusters = new HashSet<>(targetClusters);
-        final boolean includeLocal = remoteClusters.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
-        lookupPolicies(remoteClusters, includeLocal, unresolvedPolicies, listener.map(lookupResponses -> {
+        final Map<String, Boolean> remotes = new HashMap<>(targetClusters);
+        final boolean includeLocal = targetClusters.containsKey(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+        remotes.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+        lookupPolicies(remotes.keySet(), includeLocal, unresolvedPolicies, listener.map(lookupResponses -> {
             final EnrichResolution enrichResolution = new EnrichResolution();
 
             Map<String, LookupResponse> lookupResponsesToProcess = new HashMap<>();
 
             for (Map.Entry<String, LookupResponse> entry : lookupResponses.entrySet()) {
                 String clusterAlias = entry.getKey();
                 if (entry.getValue().connectionError != null) {
-                    enrichResolution.addUnavailableCluster(clusterAlias, entry.getValue().connectionError);
+                    enrichResolution.addUnusableRemote(clusterAlias, entry.getValue().connectionError);
                     // remove unavailable cluster from the list of clusters which is used below to create the ResolvedEnrichPolicy
-                    remoteClusters.remove(clusterAlias);
+                    Boolean removedVal = remotes.remove(clusterAlias);
+                    assert removedVal != null : "Remote " + clusterAlias + " was not removed from the remotes list.";
```

Comment:
Not sure I understand this assert. What is it supposed to check for?

Reply:
It's to remove a cluster from remotes since it's unusable and is same as:
```java
if (remotes.contains(clusterAlias)) {
    remotes.remove(clusterAlias);
} else {
    // clusterAlias was not removed from the remotes list.
}
```

`remove()` returns the value that's removed and writing it the other way simply avoids a double lookup, I believe? If a cluster is unusable due to a conn error but couldn't be removed from `remotes` (because it doesn't exist?), something isn't adding up.

Reply:
The map remove method returns the last value, not a boolean of whether it was successfully removed. Pawan's example is correct, but I want it as an assert to ensure that we are actually removing it, as a sanity check while testing/running snapshot builds. It is essential that the cluster be removed from this list in order to meet the requirements of this PR.